pkg/arvo/sys/vane/gall.hoon

```diff
@@ -1110,31 +1110,32 @@
     ++  ap-ducts-from-paths
       |=  [target-paths=(list path) target-ship=(unit ship)]
       ^-  (list duct)
-      ?:  &(?=(~ target-paths) ?=(~ target-ship))
-        ~[agent-duct]
+      ?~  target-paths
+        ?~  target-ship
+          ~[agent-duct]
+        %+  murn  ~(tap by incoming.subscribers.current-agent)
+        |=  [=duct =ship =path]
+        ^-  (unit ^duct)
+        ?:  =(target-ship `ship)
+          `duct
+        ~
       %-  zing
       %+  turn  target-paths
       |=  =path
-      (ap-ducts-from-path `path target-ship)
+      (ap-ducts-from-path path target-ship)
     ::  +ap-ducts-from-path: get ducts subscribed to path
     ::
     ++  ap-ducts-from-path
-      |=  [target-path=(unit path) target-ship=(unit ship)]
+      |=  [target-path=path target-ship=(unit ship)]
       ^-  (list duct)
-      ?:  &(?=(~ target-path) ?=(~ target-ship))
-        ~[agent-duct]
       %+  murn  ~(tap by incoming.subscribers.current-agent)
       |=  [=duct =ship =path]
       ^-  (unit ^duct)
       ?~  target-ship
-        ?:  =(target-path `path)
-          `duct
-        ~
-      ?~  target-path
-        ?:  =(target-ship `ship)
+        ?:  =(target-path path)
           `duct
         ~
-      ?:  &(=(target-path `path) =(target-ship `ship))
+      ?:  &(=(target-path path) =(target-ship `ship))
```

Comment:
I think these conditionals could be merged now; the desired logic is: "if target-path is path AND (target-ship is ship OR target-ship is ~)". Is there a unit logic helper that can express this concisely?

Reply:
Not a commonly used one, and this is complex enough that I'd optimize for readability.  I think what you have here is fine, the other option that comes to mind is a tall-form `?&` with wide-form `?|`.  That could be more readable.

Reply:
ok, let's go with that. I was thinking something like:
```
      ?:  &(=(target-path path) =((fall target-ship ship) ship))
        `duct
      ~
```
but it's probably best to be explicit about checking for `~`.

Reply:
Oh, I guess that would work.  For me, the explicit check is clearer, but either is fine.